### ISSUE TITLE
When Error Prone is placed in bootclasspath, classes being compiled can reference Error Prone's classes

### ISSUE THREAD

**Issue by msridhar** [google/error-prone#934]
This is with EP 2.2.0, running on JDK 1.8.0_144.  Input file:
```java
public class Test {
  public static void main(String[] args) {
    System.out.println(com.google.common.collect.ImmutableSet.class);
  }
}
```

Compilation command:
```
java -Xbootclasspath/p:error_prone_ant-2.2.0.jar com.google.errorprone.ErrorProneCompiler Test.java
```

I expected a compilation error due to the reference to `ImmutableSet`.  Instead, the code compiles successfully.  Further, it's not just an issue of Error Prone itself being in the bootclasspath.  The following command also succeeds:
```
java -Xbootclasspath/p:javac-9+181-r4173-1.jar -classpath error_prone_ant-2.2.0.jar com.google.errorprone.ErrorProneCompiler Test.java
```
The following variant with `-classpath .` fails:
```
java -Xbootclasspath/p:javac-9+181-r4173-1.jar -classpath error_prone_ant-2.2.0.jar com.google.errorprone.ErrorProneCompiler -classpath . Test.java
```
It seems javac itself has some similar behavior, e.g., this succeeds:
```
javac -Xbootclasspath/p:guava-24.0-jre.jar -classpath . Test.java
```
Assuming it's not that easy to fix this leakage of the bootclasspath, it might be good to qualify in the Java 8 section of [the docs](http://errorprone.info/docs/installation) the risks of sticking extra classes in the bootclasspath when running Error Prone.

**Comment by cushon** [comment]
The preferred way to run Error Prone at this point is the `-Xplugin` integration, which puts Error Prone on the `-processorpath` and not the `-bootclasspath`, and should avoid leaking Error Prone internals onto the compilation classpath.
